x-pack/auditbeat/module/system/host/host.go

```diff
@@ -55,17 +78,19 @@ type Host struct {
 func (host *Host) changeDetectionHash() uint64 {
 	h := xxhash.New64()
 
-	mapstr := host.toMapStr()
-
-	// We detect changes to the ID separately, so ignore here.
-	mapstr.Delete("id")
+	if host.info.Containerized != nil {
+		h.WriteString(strconv.FormatBool(*host.info.Containerized))
+	}
 
-	// Changes in uptime and boottime indicate a system reboot,
-	// which we report separately.
-	mapstr.Delete("uptime")
-	mapstr.Delete("boottime")
+	h.WriteString(host.info.Timezone)
+	h.WriteString(strconv.Itoa(host.info.TimezoneOffsetSec))
```

Comment:
Rather than converting the number to a string (which requires allocation) I think you can use

`binary.Write(h, binary.BigEndian, int32(host.Info.TimezoneOffsetSec))`

which will write the bytes directly into the hash. It should not fallback to reflection if you cast from `int` to `int32`.

Reply:
Makes sense. Does it matter if it writes big-endian or little-endian? I see us using both quite heavily in the Beats code.

Reply:
Endian doesn’t matter. It only matters that it’s hard-coded so that the hash doesn’t change between architectures.